internal/http/check.go

```diff
@@ -0,0 +1,210 @@
+			w.Write([]byte(`"OK"`))
+		} else {
+			// the server is shutting down
+			http.Error(w, "shuttingDown", http.StatusServiceUnavailable)
+		}
+	})
+	http.HandleFunc("/run", c.ServeHTTP)
+	c.Logger.Info(fmt.Sprintf("Listening at %s", c.server.Addr))
+	go func() {
+		if err := c.server.ListenAndServe(); !errors.Is(err, http.ErrServerClosed) {
+			c.Logger.WithError(err).Error("Starting http server")
+		}
+		c.Logger.Info("Stopped serving new connections.")
+		if c.shuttedDown != nil {
+			c.Logger.Info("Notifying shuttedDown")
+			c.shuttedDown <- 1
+			c.Logger.Info("Notified shuttedDown")
+		}
+	}()
+
+	s := <-c.exitSignal
+
+	c.Logger.WithField("signal", s.String()).Info("Stopping server")
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second) // TODO: Allow configure value.
+	defer cancel()
+	if err := c.server.Shutdown(ctx); err != nil {
+		c.Logger.WithError(err).Error("Shutting down server")
+	}
+	c.Logger.Info("Finished RunAndServe.")
+}
+
+type Job struct {
+	CheckID      string            `json:"check_id"`      // Required
+	StartTime    time.Time         `json:"start_time"`    // Required
+	Image        string            `json:"image"`         // Required
+	Target       string            `json:"target"`        // Required
+	Timeout      int               `json:"timeout"`       // Required
+	AssetType    string            `json:"assettype"`     // Optional
+	Options      string            `json:"options"`       // Optional
+	RequiredVars []string          `json:"required_vars"` // Optional
+	Metadata     map[string]string `json:"metadata"`      // Optional
+	RunTime      int64
+}
+
+// Shutdown is needed to fulfil the check interface and in this case we are
+// shutting down the http server and waiting
+func (c *Check) Shutdown() error {
+	// Send the exit signal to shutdown the server.
+	c.exitSignal <- syscall.SIGTERM
+
+	c.shuttedDown = make(chan int)
+	// Wait for the server to shutdown.
+	c.Logger.Info("Shutdown: waiting for shuttedDown")
+	<-c.shuttedDown
+	c.Logger.Info("Shutdown:shutted down")
+	return nil
+}
+
+// NewCheck creates new check to be run from the command line without having an agent.
+func NewCheck(name string, checker Checker, logger *log.Entry, conf *config.Config) *Check {
+	c := &Check{
+		Name:        name,
+		Logger:      logger,
+		config:      conf,
+		exitSignal:  make(chan os.Signal, 1),
+		port:        *conf.Port,
+		shuttedDown: nil,
+	}
+	c.server = &http.Server{Addr: fmt.Sprintf(":%d", c.port)}
+	signal.Notify(c.exitSignal, syscall.SIGINT, syscall.SIGTERM)
+	c.checker = checker
+	return c
+}
+
+// State holds the state for a local check.
+type State struct {
+	state agent.State
+}
+
+// Checker defines the shape a checker must have in order to be executed as vulcan-check.
+type Checker interface {
+	Run(ctx context.Context, target, assetType, opts string, state state.State) error
+	CleanUp(ctx context.Context, target, assetType, opts string)
+}
```

Comment:
Isn't this interface duplicated in `internal/http`, `internal/push` and `internal/local`? Why?

Reply:
and in bootstrapper, but I only did once.

Reply:
Again. This has to change after the rewrite.